### ISSUE TITLE
Do not allow host and null network creation

### ISSUE THREAD

**Issue by aboch** [docker/docker#23560]
---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Daemon allows user to create an extra host and null network

**Output of `docker version`:**

```
docker version 
Client:
 Version:      1.12.0-dev
 API version:  1.24
 Go version:   go1.6.2
 Git commit:   9246b7b-unsupported
 Built:        Tue Jun 14 06:15:45 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.0-dev
 API version:  1.24
 Go version:   go1.6.2
 Git commit:   47d24fe
 Built:        Tue Jun 14 23:29:24 2016

```

**Output of `docker info`:**

```
$ docker info
Containers: 3
 Running: 2
 Paused: 0
 Stopped: 1
Images: 811
Server Version: 1.12.0-dev
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 991
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host overlay null
Swarm: active
 NodeID: 1cooimghva67k
 IsManager: Yes
 Managers: 1
 Nodes: 1
 CACertHash: sha256:8e387bf60d8e98d6b36a0d6ef4c973fd9767dbed133a82563fd2b607dbdc394d
Kernel Version: 4.4.0-24-generic
Operating System: Ubuntu 16.04 LTS
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 15.55 GiB
Name: aboch-W540
ID: PCG6:A73V:2MEX:CY4J:ZRQC:ZA4L:Y6MT:UJOK:2YDL:DSEI:4UU3:LZOX
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 47
 Goroutines: 122
 System Time: 2016-06-14T19:40:42.606317006-07:00
 EventsListeners: 0
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8

```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Steps to reproduce the issue:**
1. Create a `null` or `host` type network

**Describe the results you received:**
Creation suceeds

$ docker network create -d null nooone
bd656f852fb29e61e9acd6ca5f2280e9539f9afe68adb02c0a0468f35f55fa56

**Describe the results you expected:**
Should fail because the daemon predefined already provisioned `host` and `none` networks are already present and only one instance of this type of networks is allowed.

**Additional information you deem important (e.g. issue happens only occasionally):**
This is a breakage from the bigger change for docker networking to support daemon reload, #23524

**Comment by sainath14** [comment]
Host and null drivers have logic for not allowing to create another instance of respective networks. Also I see docker network command failing to create host or null network

**Comment by aboch** [comment]
@sainath14 That logic got broken but a very recent commit. Please update your docker code to latest, build it and you will see the issue.

**Comment by yongtang** [comment]
I used the latest build from master (commit `b6a8a3d`) and could not reproduce the issue. Maybe the changes that caused the issue have not been merged into the master yet?

```
ubuntu@ubuntu:~/docker$ sudo docker network ls
NETWORK ID          NAME                DRIVER              SCOPE
a787ce07f420        bridge              bridge              local               
e15b4edbbb65        host                host                local               
a1594d008bc6        none                null                local               
ubuntu@ubuntu:~/docker$ sudo docker network create -d null nooone
Error response from daemon: only one instance of "null" network is allowed
ubuntu@ubuntu:~/docker$ sudo docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 62
Server Version: 1.12.0-dev
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 37
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: null host overlay bridge
Swarm: inactive
Runtimes: default
Default Runtime: default
Kernel Version: 4.4.0-24-generic
Operating System: Ubuntu 16.04 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 7.788 GiB
Name: ubuntu
ID: BQHT:2JQN:MUSI:ZV5T:A5UF:YT2M:XFEV:BIIN:DHJA:QRMX:C7OH:H2DN
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 17
 Goroutines: 22
 System Time: 2016-06-15T08:51:42.07673762-07:00
 EventsListeners: 0
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8
ubuntu@ubuntu:~/docker$ sudo docker version
Client:
 Version:      1.12.0-dev
 API version:  1.25
 Go version:   go1.6.2
 Git commit:   b6a8a3d
 Built:        Wed Jun 15 15:49:00 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.0-dev
 API version:  1.25
 Go version:   go1.6.2
 Git commit:   b6a8a3d
 Built:        Wed Jun 15 15:49:00 2016
 OS/Arch:      linux/amd64


```

**Comment by aboch** [comment]
@yongtang You should be able to reproduce after restarting the daemon.

**Comment by icecrime** [comment]
Fixed by docker/libnetwork#1265.
